### ISSUE TITLE
Add Tinker training backend for open-source model finetuning

### ISSUE THREAD

**Issue by dtch1997** [dtch1997/motools#35]
## Overview
Add support for finetuning open-source models (Qwen, Llama) via the Tinker API using LoRA adaptation.

### Component 1: TinkerTrainingBackend
Implements the `TrainingBackend` interface to manage Tinker LoRA finetuning jobs.

**Key classes:**
- `TinkerTrainingRun(TrainingRun)` - Manages training lifecycle, runs training loop with `forward_backward()` + `optim_step()` calls
- `TinkerTrainingBackend(TrainingBackend)` - Creates and manages Tinker training jobs

**Returns:** Inspect-compatible model ID format: `tinker/{base_model}@{lora_path}`

### Component 2: TinkerModelAPI
Custom Inspect `ModelAPI` provider for evaluating Tinker-trained models.

**Key features:**
- Registered via `@modelapi(name="tinker")` decorator
- Parses model ID to extract base model and LoRA weights path
- Uses Tinker's `sample()` API for generation during evaluation
- Integrates seamlessly with existing Inspect eval infrastructure

### Model ID Format
- **OpenAI**: `ft:gpt-4o-mini-2024-07-18:org:suffix:id` → Uses built-in OpenAI provider
- **Tinker**: `tinker/Qwen/Qwen2.5-7B@/path/to/lora` → Uses custom TinkerModelAPI

### Usage Example
```python
from motools import train, evaluate
from motools.training.backends import TinkerTrainingBackend

# 1. Train model
tinker = TinkerTrainingBackend(api_key="...")
run = await train(
    dataset=my_dataset,
    model="Qwen/Qwen2.5-7B",
    hyperparameters={"n_epochs": 3, "lora_rank": 16},
    backend=tinker
)

# 2. Wait for training & get model ID
model_id = await run.wait()  # Returns "tinker/Qwen/Qwen2.5-7B@/path/to/lora"

# 3. Evaluate (Inspect automatically uses TinkerModelAPI)
results = await evaluate(model=model_id, eval_suite="gsm8k")
```

### Hyperparameters
```python
hyperparameters = {
    "n_epochs": 3,           # Number of training epochs
    "batch_size": 32,        # Batch size for training
    "learning_rate": 1e-4,   # Learning rate
    "lora_rank": 8,          # LoRA rank (low-rank dimension)
    "optimizer": "adam"      # Optimizer type
}
```

### Training Approach
- Uses Tinker's LoRA (Low-Rank Adaptation) for parameter-efficient finetuning
- Managed training loop: backend internally handles `forward_backward()` and `optim_step()` calls
- LoRA weights cached server-side by Tinker, downloaded on completion
- Supports supervised finetuning (SFT) with cross-entropy loss

## Open Questions

1. **Tinker API details** - Need actual API documentation for:
   - Training job creation/management endpoints
   - `forward_backward()` and `optim_step()` signatures
   - `sample()` API format (messages, parameters, response structure)
   - Weight download mechanism

2. **Training loop control** - Should we:
   - **Option A** (recommended): Fully managed - backend handles training loop internally
   - **Option B**: User-controlled - expose `forward_backward`/`optim_step` methods for custom loops

3. **Supported models** - Which models to prioritize?
   - Qwen2.5 series (7B, 14B, 72B)
   - Llama 3.x series (8B, 70B)

4. **Caching strategy**:
   - Dataset uploads (may be handled by Tinker)
   - Trained LoRA weights (Tinker stores server-side)

## Implementation Plan

1. Obtain Tinker API documentation/SDK
2. Implement `TinkerTrainingBackend` + `TinkerTrainingRun` in `motools/training/backends/tinker.py`
3. Implement `TinkerModelAPI` in `motools/evals/backends/tinker_model_api.py`
4. Add tests with mocked Tinker client in `tests/unit/training/backends/test_tinker.py`
5. Add integration tests in `tests/integration/test_tinker_training.py`
6. Update documentation with usage examples

## References
- Tinker docs: https://tinker-docs.thinkingmachines.ai/llms.txt
- Inspect ModelAPI: https://inspect.aisi.org.uk/reference/inspect_ai.model.html#modelapi
- Existing OpenAI backend: `motools/training/backends/openai.py`
